mantidimaging/gui/windows/spectrum_viewer/spectrum_widget.py

```diff
@@ -353,3 +354,47 @@ def __init__(self, main_window: MainWindowView) -> None:
         nan_check_menu = [("Crop Coordinates", lambda: main_window.presenter.show_operation("Crop Coordinates")),
                           ("NaN Removal", lambda: main_window.presenter.show_operation("NaN Removal"))]
         self.image.enable_nan_check(actions=nan_check_menu)
+
+
+class MIPlotItem(PlotItem):
+    connect: str | np.ndarray = 'auto'
+    pen_dict: dict = {}
+    name_index: int = 0
+
+    def __init__(self, join_plot=True, *args, **kwds) -> None:
```

Comment:
I don't think the `join_plot` argument does anything. Maybe easier just to drop it. And for in `set_join_plot()` have `join_plot` as a local argument.